rust/datafusion/tests/user_defined_plan.rs

```diff
@@ -471,12 +471,10 @@ impl Stream for TopKReader {
             return Poll::Ready(None);
         }
         // this aggregates and thus returns a single RecordBatch.
-        self.done = true;
```

Comment:
this is probably a real bug -- the topK isn't actually done until it produces output -- and the way this code is written `top_values.poll_next_unpin()` may not return `Poll::NotReady` ? Maybe?

Anyhow, looks great

Reply:
That was my intuition too, yeah. This seems to be more correct and fixes the issue.